iib/web/models.py

```diff
@@ -658,6 +662,80 @@ def to_json(self, verbose=True):
         return rv
 
 
+class RequestRegenerateBundle(Request):
+    """A "regenerate_bundle" image build request."""
+
+    __tablename__ = 'request_regenerate_bundle'
+
+    id = db.Column(db.Integer, db.ForeignKey('request.id'), autoincrement=False, primary_key=True)
+    # The ID of the regenerated bundle image
+    bundle_image_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=True)
+    bundle_image = db.relationship('Image', foreign_keys=[bundle_image_id], uselist=False)
+    # The ID of the bundle image to be regenerated
+    from_bundle_image_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=False)
+    from_bundle_image = db.relationship('Image', foreign_keys=[from_bundle_image_id], uselist=False)
+    # The ID of the resolved bundle image to be regenerated
+    from_bundle_image_resolved_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=True)
+    from_bundle_image_resolved = db.relationship(
+        'Image', foreign_keys=[from_bundle_image_resolved_id], uselist=False
+    )
+    # The name of the organization the bundle should be regenerated for
+    organization = db.Column(db.String, nullable=True)
+
+    __mapper_args__ = {
+        'polymorphic_identity': RequestTypeMapping.__members__['regenerate_bundle'].value,
+    }
+
+    @classmethod
+    def from_json(cls, kwargs):
+        """Handle JSON requests for the Regenerate Bundle API endpoint."""
+        request_kwargs = deepcopy(kwargs)
+
+        validate_request_params(
+            request_kwargs, required_params={'from_bundle_image'}, optional_params={'organization'},
+        )
+
+        # Validate organization is correctly provided
+        organization = request_kwargs.get('organization')
+        if organization and not isinstance(organization, str):
+            raise ValidationError(f'"organization" must be a string')
+
+        # Validate from_bundle_image is correctly provided
+        from_bundle_image = request_kwargs.pop('from_bundle_image')
+        if not isinstance(from_bundle_image, str):
+            raise ValidationError('"from_bundle_image" must be a string')
+
+        request_kwargs['from_bundle_image'] = Image.get_or_create(
+            pull_specification=from_bundle_image
+        )
+
+        # current_user.is_authenticated is only ever False when auth is disabled
```

Comment:
Does it make sense to move this before you add the from_bundle_image param to the db? That would avoid unnecessary database operations. Thoughts?

Reply:
I don't understand the question. If the instance of IIB does not have authentication enabled, then `user` is not set. That's a valid scenario, e.g. local dev environment with docker-compose. In either case, the same amount of database operations are performed.

Reply:
my bad, I misinterpreted something. If login is enabled and the user is not authenticated, it wouldn't reach here in the first place.